src/Vault.sol

```diff
@@ -460,6 +460,31 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP {
         return this.onERC721Received.selector;
     }
 
+    function decreaseLiquidity(INonfungiblePositionManager.DecreaseLiquidityParams calldata params)
+        external
+        returns (uint256 amount0, uint256 amount1)
+    {
+        _requireVaultOwner(vaultIdByNft[params.tokenId]);
+
+        (amount0, amount1) = positionManager.decreaseLiquidity(params);
+    }
+
+    function collect(INonfungiblePositionManager.CollectParams calldata params)
+        external
+        returns (uint256 amount0, uint256 amount1)
+    {
+        uint256 vaultId = vaultIdByNft[params.tokenId];
+        _requireVaultOwner(vaultId);
+
+        (amount0, amount1) = positionManager.collect(params);
+
+        // checking that health factor is more or equal than 1
+        (, uint256 adjustedCollateral) = calculateVaultCollateral(vaultId);
+        if (adjustedCollateral < getOverallDebt(vaultId)) {
```

Comment:
There was also a `minSingleNftCollateral` setting, we need to do something about it. Choices are:
1. Ignore it here. In this case this setting will not have a sense at all, since user can always do a deposit+collect to bypass it.
2. Remove `minSingleNftCollateral` (or replace it with `minDebtPerVault` ?), maybe `maxDebtPerVault` and `maxNftsPerVault` limits alone would be good enough.
3. Don't allow to collect tokens from position, if it lowers the NFT value below `minSingleNftCollateral`. This will probably lead to a not very pleasant UX